dashboard-ui/src/components/TextBasedScenarios/TextBasedScenariosPage.jsx

```diff
@@ -59,19 +60,18 @@ class TextBasedScenariosPage extends Component {
     }
 
     introSurveyComplete = (survey) => {
-        console.log(survey.data)
         const scenarioOrderString = survey.data.scenarioOrder.replace(/\\/g, "");
         const scenarioOrderArray = JSON.parse(scenarioOrderString);
-        console.log(scenarioOrderArray)
         this.setState({
             scenarios: scenarioOrderArray,
             participantID: survey.data["Participant ID"],
             vrEnvCompleted: survey.data["vrEnvironmentsCompleted"]
         })
 
-        const selectedScenarios = []
+        let selectedScenarios = []
         for (const scenario of scenarioOrderArray) {
-            selectedScenarios.push(scenarioMappings[scenario])
+            // make deep copies of json files to be sure the originals are not unintentionally tampered with
+            selectedScenarios.push(JSON.parse(JSON.stringify(scenarioMappings[scenario])))
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/structuredClone

I wonder if you'd be able to use this here?  It's pretty new, I just learned about it last year, but it cleans up the whole JSON parse/stringify method of cloning

Reply:
Thanks, I'll bookmark this page and be sure to test that and use it in my next PR. Just don't want to accidentally break something right now